From 20c946a20213274de3bed4422672e72c7cfdbbe6 Mon Sep 17 00:00:00 2001
From: Rahul Ganesh <rahulgab@amazon.com>
Date: Thu, 22 Jan 2026 16:46:18 -0800
Subject: [PATCH 1/5] Remove secondstar service

EKS-A does not use the secondstar (SSH over serial) service. Remove it
to reduce binary size and dependencies.

Note: tootles (metadata service, formerly hegel) is kept as EKS-A uses
it for machine metadata.

Signed-off-by: Rahul Ganesh <rahulgab@amazon.com>
---
 cmd/tinkerbell/cmd.go         | 48 ++---------------------------------
 cmd/tinkerbell/flag/global.go |  7 -----
 2 files changed, 2 insertions(+), 53 deletions(-)

diff --git a/cmd/tinkerbell/cmd.go b/cmd/tinkerbell/cmd.go
index 4b7d3a6a..c9dc0452 100644
--- a/cmd/tinkerbell/cmd.go
+++ b/cmd/tinkerbell/cmd.go
@@ -23,7 +23,6 @@ import (
 	"github.com/tinkerbell/tinkerbell/pkg/backend/kube"
 	"github.com/tinkerbell/tinkerbell/pkg/build"
 	"github.com/tinkerbell/tinkerbell/rufio"
-	"github.com/tinkerbell/tinkerbell/secondstar"
 	"github.com/tinkerbell/tinkerbell/smee"
 	"github.com/tinkerbell/tinkerbell/tink/controller"
 	"github.com/tinkerbell/tinkerbell/tink/server"
@@ -37,7 +36,6 @@ import (
 const (
 	defaultRufioMetricsPort          = 8082
 	defaultRufioProbePort            = 8083
-	defaultSecondStarPort            = 2222
 	defaultSmeeHTTPPort              = 7171
 	defaultSmeeHTTPSPort             = 7272
 	defaultTinkControllerMetricsPort = 8080
@@ -63,7 +61,6 @@ func Execute(ctx context.Context, cancel context.CancelFunc, args []string) erro
 		EnableTinkServer:     true,
 		EnableTinkController: true,
 		EnableRufio:          true,
-		EnableSecondStar:     true,
 		EnableUI:             true,
 		EnableCRDMigrations:  true,
 		EmbeddedGlobalConfig: flag.EmbeddedGlobalConfig{
@@ -116,14 +113,6 @@ func Execute(ctx context.Context, cancel context.CancelFunc, args []string) erro
 		Config: rufio.NewConfig(rufioOpts...),
 	}
 
-	ssc := &flag.SecondStarConfig{
-		Config: &secondstar.Config{
-			SSHPort:      defaultSecondStarPort,
-			IPMITOOLPath: "/usr/sbin/ipmitool",
-			IdleTimeout:  15 * time.Minute,
-		},
-	}
-
 	uiOpts := []ui.Option{
 		ui.WithURLPrefix("/"),
 		ui.WithBindPort(defaultUIPort),
@@ -142,7 +131,6 @@ func Execute(ctx context.Context, cancel context.CancelFunc, args []string) erro
 	tfs := ff.NewFlagSet("tink server - Workflow service").SetParent(hfs)
 	cfs := ff.NewFlagSet("tink controller - Workflow controller").SetParent(tfs)
 	rfs := ff.NewFlagSet("rufio - BMC controller").SetParent(cfs)
-	ssfs := ff.NewFlagSet("secondstar - SSH over serial service").SetParent(rfs)
 	uifs := ff.NewFlagSet("ui - UI service").SetParent(ssfs)
 	gfs := ff.NewFlagSet("globals").SetParent(uifs)
 	flag.RegisterSmeeFlags(&flag.Set{FlagSet: sfs}, s)
@@ -150,7 +138,6 @@ func Execute(ctx context.Context, cancel context.CancelFunc, args []string) erro
 	flag.RegisterTinkServerFlags(&flag.Set{FlagSet: tfs}, ts)
 	flag.RegisterTinkControllerFlags(&flag.Set{FlagSet: cfs}, tc)
 	flag.RegisterRufioFlags(&flag.Set{FlagSet: rfs}, rc)
-	flag.RegisterSecondStarFlags(&flag.Set{FlagSet: ssfs}, ssc)
 	flag.RegisterUIFlags(&flag.Set{FlagSet: uifs}, uic)
 	flag.RegisterGlobal(&flag.Set{FlagSet: gfs}, globals)
 	if embeddedApiserverExecute != nil && embeddedFlagSet != nil {
@@ -183,7 +170,6 @@ func Execute(ctx context.Context, cancel context.CancelFunc, args []string) erro
 		"tinkServerEnabled", globals.EnableTinkServer,
 		"tinkControllerEnabled", globals.EnableTinkController,
 		"rufioEnabled", globals.EnableRufio,
-		"secondStarEnabled", globals.EnableSecondStar,
 		"uiEnabled", globals.EnableUI,
 		"publicIP", globals.PublicIP,
 		"embeddedKubeAPIServer", globals.EmbeddedGlobalConfig.EnableKubeAPIServer,
@@ -238,14 +224,6 @@ func Execute(ctx context.Context, cancel context.CancelFunc, args []string) erro
 		rc.Config.ProbeAddr = netip.AddrPortFrom(globals.BindAddr, rc.Config.ProbeAddr.Port())
 	}
 
-	// Second star
-	if err := ssc.Convert(); err != nil {
-		return fmt.Errorf("failed to convert secondstar config: %w", err)
-	}
-	if globals.BindAddr.IsValid() {
-		ssc.Config.BindAddr = globals.BindAddr
-	}
-
 	// UI
 	uic.Convert(globals.BindAddr, globals.TLS.CertFile, globals.TLS.KeyFile)
 
@@ -311,7 +289,7 @@ func Execute(ctx context.Context, cancel context.CancelFunc, args []string) erro
 			cliLog.Info("CRD migrations completed")
 		}
 
-		b, err := newKubeBackend(ctx, globals.BackendKubeConfig, "", globals.BackendKubeNamespace, enabledIndexes(globals.EnableSmee, globals.EnableTootles, globals.EnableTinkServer, globals.EnableSecondStar), WithQPS(globals.BackendKubeOptions.QPS), WithBurst(globals.BackendKubeOptions.Burst))
+		b, err := newKubeBackend(ctx, globals.BackendKubeConfig, "", globals.BackendKubeNamespace, enabledIndexes(globals.EnableSmee, globals.EnableTootles, globals.EnableTinkServer), WithQPS(globals.BackendKubeOptions.QPS), WithBurst(globals.BackendKubeOptions.Burst))
 		if err != nil {
 			return fmt.Errorf("failed to create kube backend: %w", err)
 		}
@@ -324,7 +302,6 @@ func Execute(ctx context.Context, cancel context.CancelFunc, args []string) erro
 		tc.Config.Client = b.ClientConfig
 		tc.Config.DynamicClient = b
 		rc.Config.Client = b.ClientConfig
-		ssc.Config.Backend = b
 	case "file":
 		b, err := newFileBackend(ctx, log, globals.BackendFilePath)
 		if err != nil {
@@ -418,19 +395,6 @@ func Execute(ctx context.Context, cancel context.CancelFunc, args []string) erro
 		return nil
 	})
 
-	// SecondStar
-	g.Go(func() error {
-		if !globals.EnableSecondStar {
-			cliLog.Info("secondstar service is disabled")
-			return nil
-		}
-		ll := ternary((ssc.LogLevel != 0), ssc.LogLevel, globals.LogLevel)
-		if err := ssc.Config.Start(ctx, getLogger(ssc.NoLog, ll).WithName("secondstar")); err != nil {
-			return fmt.Errorf("failed to start secondstar service: %w", err)
-		}
-		return nil
-	})
-
 	// UI
 	g.Go(func() error {
 		if !globals.EnableUI {
@@ -475,16 +439,13 @@ func numEnabled(globals *flag.GlobalConfig) int {
 	if globals.EnableRufio {
 		n++
 	}
-	if globals.EnableSecondStar {
-		n++
-	}
 	if globals.EnableUI {
 		n++
 	}
 	return n
 }
 
-func enabledIndexes(smeeEnabled, tootlesEnabled, tinkServerEnabled, secondStarEnabled bool) map[kube.IndexType]kube.Index {
+func enabledIndexes(smeeEnabled, tootlesEnabled, tinkServerEnabled bool) map[kube.IndexType]kube.Index {
 	idxs := make(map[kube.IndexType]kube.Index, 0)
 
 	if smeeEnabled {
@@ -500,11 +461,6 @@ func enabledIndexes(smeeEnabled, tootlesEnabled, tinkServerEnabled, secondStarEn
 			idxs[k] = v
 		}
 	}
-	if secondStarEnabled {
-		for k, v := range flag.KubeIndexesSecondStar {
-			idxs[k] = v
-		}
-	}
 
 	return idxs
 }
diff --git a/cmd/tinkerbell/flag/global.go b/cmd/tinkerbell/flag/global.go
index 30fba8ec..ce112939 100644
--- a/cmd/tinkerbell/flag/global.go
+++ b/cmd/tinkerbell/flag/global.go
@@ -23,7 +23,6 @@ type GlobalConfig struct {
 	EnableTinkServer     bool
 	EnableTinkController bool
 	EnableRufio          bool
-	EnableSecondStar     bool
 	EnableUI             bool
 	EnableCRDMigrations  bool
 	EmbeddedGlobalConfig EmbeddedGlobalConfig
@@ -59,7 +58,6 @@ func RegisterGlobal(fs *Set, gc *GlobalConfig) {
 	fs.Register(EnableTinkServer, ffval.NewValueDefault(&gc.EnableTinkServer, gc.EnableTinkServer))
 	fs.Register(EnableTinkController, ffval.NewValueDefault(&gc.EnableTinkController, gc.EnableTinkController))
 	fs.Register(EnableRufioController, ffval.NewValueDefault(&gc.EnableRufio, gc.EnableRufio))
-	fs.Register(EnableSecondStar, ffval.NewValueDefault(&gc.EnableSecondStar, gc.EnableSecondStar))
 	fs.Register(EnableUI, ffval.NewValueDefault(&gc.EnableUI, gc.EnableUI))
 	fs.Register(EnableCRDMigrations, ffval.NewValueDefault(&gc.EnableCRDMigrations, gc.EnableCRDMigrations))
 	fs.Register(LogLevelConfig, ffval.NewValueDefault(&gc.LogLevel, gc.LogLevel))
@@ -162,11 +160,6 @@ var EnableRufioController = Config{
 	Usage: "enable Rufio Controller service",
 }
 
-var EnableSecondStar = Config{
-	Name:  "enable-secondstar",
-	Usage: "enable SecondStar service",
-}
-
 var EnableUI = Config{
 	Name:  "enable-ui",
 	Usage: "enable UI service",
-- 
2.49.0

